src/teacher/commands/list-teacher-subjects.command.ts

```diff
@@ -18,6 +18,7 @@ export class ListTeacherSubjectsCommand {
           include: {
             subject: {
               include: {
+                studentsEnrolled: true,
```

Comment:
Por que você alterou esse comando?

Reply:
Pra trazer essa informação dentro do `SubjectFactory`. 

```
const teacher = await this.prisma.teacher.findUnique({
      where: { user_id: teacherId },
      include: {
        user: true,
        SubjectResponsability: {
          include: {
            subject: {
              include: {
                studentsEnrolled: true,
                course: true,
              },
            },
          },
        },
      },
    });

    if (!teacher) {
      throw new TeacherNotFoundException();
    }

    return teacher.SubjectResponsability.map((sub) =>
      SubjectFactory.createSubject(sub.subject),
    );
  }
```
Se não incluir ele na query, ele não traz essa info pra gente dentro do subject que a factory vai usar. E ce tinha pedido pra modificar o factory tbm.

Reply:
Mas não precisa mudar as rotas, só deixar a factory pronta pra quando for necessário